service/fact-check/action/claim/list.go

```diff
@@ -39,14 +45,61 @@ func list(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// Filters
+	filterRatingID := r.URL.Query().Get("rating")
+	filterClaimantID := r.URL.Query().Get("claimant")
+	searchQuery := r.URL.Query().Get("q")
+	sort := r.URL.Query().Get("sort")
+
+	filters := generateFilters(filterRatingID, filterClaimantID)
+	filteredClaimIDs := make([]uint, 0)
+
+	if filters != "" {
+		filters = fmt.Sprint(filters, " AND space_id=", sID)
+
+		// Search claims with filter
+		var hits []interface{}
+		var result map[string]interface{}
+
+		if searchQuery != "" {
+			hits, err = meili.SearchWithQuery(searchQuery, filters, "claim")
+		} else {
+			result, err = meili.SearchWithoutQuery(filters, "claim")
+			hits = result["hits"].([]interface{})
+		}
+
+		if err != nil {
+			loggerx.Error(err)
+			errorx.Render(w, errorx.Parser(errorx.InternalServerError()))
+			return
+		}
+
+		filteredClaimIDs = meili.GetIDArray(hits)
+		if len(filteredClaimIDs) == 0 {
+			loggerx.Error(err)
+			errorx.Render(w, errorx.Parser(errorx.RecordNotFound()))
+			return
+		}
+	}
+
 	result := paging{}
 	result.Nodes = make([]model.Claim, 0)
 
 	offset, limit := paginationx.Parse(r.URL.Query())
 
-	err = config.DB.Model(&model.Claim{}).Preload("Rating").Preload("Rating.Medium").Preload("Claimant").Preload("Claimant.Medium").Where(&model.Claim{
+	if sort == "" {
+		sort = "desc"
+	}
```

Comment:
Instead of checking sort is empty or not, check sort is equal to **asc** as query parameters can be anything. And assign **desc** by default to sort.